audit/conn/conn_test.go

```diff
@@ -0,0 +1,168 @@
+package conn
+
+import (
+	"context"
+	"log"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/microsoft/go-otel-audit/audit/conn/internal/server"
+	"github.com/microsoft/go-otel-audit/audit/msgs"
+
+	"github.com/kylelemons/godebug/pretty"
+)
+
+func init() {
+	log.SetFlags(log.LstdFlags | log.Lshortfile)
+}
+
+var stdMsg = msgs.Msg{
+	Type: msgs.ControlPlane,
+	Record: msgs.Record{
+		CallerIpAddress:            msgs.MustParseAddr("192.168.0.1"),
+		CallerIdentities:           map[msgs.CallerIdentityType][]msgs.CallerIdentityEntry{msgs.UPN: {{"user1@domain.com", "Description"}}},
+		OperationCategories:        []msgs.OperationCategory{msgs.UserManagement},
+		TargetResources:            map[string][]msgs.TargetResourceEntry{"ResourceType": {{"Name", "Cluster", "DataCenter", "Region"}}},
+		CallerAccessLevels:         []string{"Level1"},
+		OperationAccessLevel:       "AccessLevel",
+		OperationName:              "Operation",
+		OperationResultDescription: "ResultDescription",
+		CallerAgent:                "Agent",
+	},
+}
+
+func msgExpect(a msgs.Record, t time.Time) []any {
+	return []any{
+		"AsmAuditDP", // event name
+		[]any{
+			[]any{
+				t.Unix(), // timestamp in seconds
+				a,
+			},
+		},
+		struct{ TimeFormat string }{TimeFormat: "DateTime"}, // required by Geneva Agent
+	}
+}
+
+func TestWriters(t *testing.T) {
+	tests := []struct {
+		// desc is a description of the test.
+		desc string
+		// newServer is a function that creates a new server.
+		newServer func() (*server.AuditRecordTest, error)
+		// newConn is a function that creates a new connection to the server.
+		newConn func() (Audit, error)
+		// resetErr is true if we want to reset the connection somewhere in the test.
+		resetErr bool
+	}{
+		{
+			desc: "domain socket",
+			newServer: func() (*server.AuditRecordTest, error) {
+				return server.New("unix", "/tmp/audit.sock")
```

Comment:
extremely minor, feel free to ignore: You could pull this `/tmp/audit.sock` into a function constant to avoid repeating it the ~3 or so times you repeat it.

Reply:
Done